docs/static/logging.asciidoc

```diff
@@ -126,10 +83,78 @@ Example response:
+immediately and do not need a restart. 
+
+NOTE: By default, the logging API attempts to bind to `tcp:9600`. If this port is already in use by another Logstash
+instance, you need to launch Logstash with the `--http.port` flag specified to bind to a different port. See
+<<command-line-flags>> for more information.
+
+===== Update logging levels
+
+Prepend the name of the subsystem, module, or plugin with `logger.`. 
+
+Here is an example using `outputs.elasticsearch`:
+
+[source,js]
+--------------------------------------------------
+curl -XPUT 'localhost:9600/_node/logging?pretty' -H 'Content-Type: application/json' -d'
+{
+    "logger.logstash.outputs.elasticsearch" : "DEBUG"
+}
+'
+--------------------------------------------------
+
+While this setting is in effect, Logstash emits DEBUG-level logs for __all__ the Elasticsearch outputs
+specified in your configuration. Please note this new setting is transient and will not survive a restart.
+
+NOTE: If you want logging changes to persist after a restart, add them to `log4j2.properties` instead. 
+
+===== Reset dynamic logging levels
+
 To reset any logging levels that may have been dynamically changed via the logging API, send a `PUT` request to
 `_node/logging/reset`. All logging levels will revert to the values specified in the `log4j2.properties` file.
 
 [source,js]
 --------------------------------------------------
 curl -XPUT 'localhost:9600/_node/logging/reset?pretty'
 --------------------------------------------------
+
+==== Log file location
+
+You can specify the log file location using `--path.logs` setting.
+
+==== Slowlog
+
+Slow-log for Logstash adds the ability to log when a specific event takes an abnormal amount of time to make its way
+through the pipeline. Just like the normal application log, you can find slow-logs in your `--path.logs` directory.
+Slowlog is configured in the `logstash.yml` settings file with the following options:
+
+[source,yaml]
+------------------------------
+slowlog.threshold.warn (default: -1)
+slowlog.threshold.info (default: -1)
+slowlog.threshold.debug (default: -1)
+slowlog.threshold.trace (default: -1)
+------------------------------
+
+By default, these values are set to `-1nanos` to represent an infinite threshold where no slowlog will be invoked. These `slowlog.threshold`
+fields are configured using a time-value format which enables a wide range of trigger intervals. The positive numeric ranges
+can be specified using the following time units: `nanos` (nanoseconds), `micros` (microseconds), `ms` (milliseconds), `s` (second), `m` (minute),
+`h` (hour), `d` (day).
+
+Here is an example:
+
+[source,yaml]
+------------------------------
+slowlog.threshold.warn: 2s
+slowlog.threshold.info: 1s
+slowlog.threshold.debug: 500ms
+slowlog.threshold.trace: 100ms
+------------------------------
+
+In the above configuration, events that take longer than two seconds to be processed within a filter will be logged.
```

Comment:
maybe it's worth explaining the thresholds, i.e. that it's "longer than two seconds if log level is set to warn, 1 if level info, 0.5 for debug, etc, so that slow log is more sensitive if you're raising the log level"